test/lib/parsers_test.exs

```diff
@@ -29,14 +29,24 @@ defmodule ExAws.STS.ParsersTest do
     get_federation_token: 2
   ]
 
-  # Build tests for all actions and verifies that mock responses were parsed
-  for {action, arity} <- @actions do
-    @tag action: action
-    test "parses `:#{action}` using mock response" do
-      assert {:ok, %{body: body}} = parse_mock_response(unquote(action), unquote(arity))
-
-      for {_, value} <- body do
-        assert value && value != "", "did not parse value for response"
+  if System.get_env("SWEET_XML") == "DISABLED" do
+    for {action, arity} <- @actions do
+      test "raises missing sweet_xml error for `:#{action}`" do
+        assert_raise RuntimeError,
+                    "Dependency sweet_xml is required for role based authentication",
+                    fn -> parse_mock_response(unquote(action), unquote(arity)) end
+      end
+    end
```

Comment:
Right now, to test this, you need to run `mix clean`, then `SWEET_XML=DISABLED mix deps.get` and `SWEET_XML=DISABLED mix test test/lib/parsers_test.exs`

I realize that this is not ideal, but I could not find a way to unload and recompile `ExAWS.STS.Parsers` at runtime. If this is possible, we could find some trick to run both tests without having to clean and reinstall the deps between tests.